libraries/AP_NavEKF2/AP_NavEKF2_Logging.cpp

```diff
@@ -238,39 +251,69 @@ void NavEKF2::Log_Write_Beacon(uint64_t time_us) const
     }
 }
 
+void NavEKF2::Log_Write_Timing(uint8_t _core, uint64_t time_us) const
+{
+    // log EKF timing statistics every 5s
+    static uint32_t lastTimingLogTime_ms = 0;
+    if (AP::dal().millis() - lastTimingLogTime_ms <= 5000) {
+        return;
+    }
+    lastTimingLogTime_ms = AP::dal().millis();
+
+    struct ekf_timing timing;
+    getTimingStatistics(_core, timing);
+
+    const struct log_NKT nkt{
+        LOG_PACKET_HEADER_INIT(LOG_NKT_MSG),
+        time_us      : time_us,
+        core         : _core,
+        timing_count : timing.count,
+        dtIMUavg_min : timing.dtIMUavg_min,
+        dtIMUavg_max : timing.dtIMUavg_max,
+        dtEKFavg_min : timing.dtEKFavg_min,
+        dtEKFavg_max : timing.dtEKFavg_max,
+        delAngDT_min : timing.delAngDT_min,
+        delAngDT_max : timing.delAngDT_max,
+        delVelDT_min : timing.delVelDT_min,
+        delVelDT_max : timing.delVelDT_max,
+    };
+    AP::logger().WriteBlock(&nkt, sizeof(nkt));
+}
+
+
 void NavEKF2::Log_Write()
 {
     // only log if enabled
     if (activeCores() <= 0) {
         return;
     }
 
-    const uint64_t time_us = AP_HAL::micros64();
+    if (lastLogWrite_us == imuSampleTime_us) {
+        // vehicle is doubling up on logging
```

Comment:
Might want to explain when this happens.

Reply:
it's due to the slow/fast logging. We can fix it, but out of scope of this PR